src/main/scala/com/microsoft/sqlserver/jdbc/spark/connectors/ReliableSingleInstanceStrategy.scala

```diff
@@ -50,7 +50,7 @@ object ReliableSingleInstanceStrategy extends  DataIOStrategy with Logging {
     logInfo("write : reliable write to single instance called")
     // Initialize - create connection and cleanup existing tables if any
     val conn = createConnectionFactory(options)()
-    val stagingTableList = getStagingTableNames(appId, df.rdd.getNumPartitions)
+    val stagingTableList = getStagingTableNames(appId, options.dbtable, df.rdd.getNumPartitions)
```

Comment:
dbtable name can be a 1 part, 2 part or 3 part name. Concat will result in tables like <app-id>A.B.C#appid. IMO, this will result in a invaild table name and will throw an error.  Should "." be replaced with "_" or just take table name and not full name?

Reply:
I think fully qualified table name is being used here. In getStagingTableName(), the staging table name generated is always enclosed with brackets, so the "." would not be a problem.